src/token-group/__tests__/token-group.test.tsx

```diff
@@ -169,4 +185,64 @@ describe('TokenGroup', () => {
       expect(icon!.getElement()).toContainHTML(icons['treeview-collapse']);
     });
   });
+
+  describe('Focus management', () => {
+    test('Focus is dispatched to the next active token when non-last token is removed', () => {
+      const wrapper = renderStatefulTokenGroup({
+        items: [
+          { label: '1', dismissLabel: 'Remove 1' },
+          { label: '2', dismissLabel: 'Remove 2' },
+          { label: '3', dismissLabel: 'Remove 3', disabled: true },
+          { label: '4', dismissLabel: 'Remove 4' },
+        ],
+      });
+      wrapper.findToken(2)!.findDismiss().click();
+
+      expect(wrapper.findToken(3)!.findDismiss().getElement()).toHaveFocus();
```

Comment:
I am preferring to use indices here specifically because the purpose of the test is to check if the correct next/prev element is referenced so the indices convey it better I think.

But I will consider having both assertions for better regression.